tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -193,25 +192,21 @@ private void PropagateMetadata(ArraySegment<Span> spans)
             }
         }
 
-        /// <summary>
-        /// When receiving chunks of spans, the backend checks whether the aas.resource.id tag is present on any of the
-        /// span to decide which metric to emit (datadog.apm.host.instance or datadog.apm.azure_resource_instance one).
-        /// </summary>
-        private void DecorateWithAASMetadata(Span span)
+        private void AddAASMetadataToTraceChunk(Span span)
         {
             if (AzureAppServices.Metadata.IsRelevant)
             {
-                span.SetTag(Datadog.Trace.Tags.AzureAppServicesSiteName, AzureAppServices.Metadata.SiteName);
-                span.SetTag(Datadog.Trace.Tags.AzureAppServicesSiteKind, AzureAppServices.Metadata.SiteKind);
-                span.SetTag(Datadog.Trace.Tags.AzureAppServicesSiteType, AzureAppServices.Metadata.SiteType);
-                span.SetTag(Datadog.Trace.Tags.AzureAppServicesResourceGroup, AzureAppServices.Metadata.ResourceGroup);
-                span.SetTag(Datadog.Trace.Tags.AzureAppServicesSubscriptionId, AzureAppServices.Metadata.SubscriptionId);
-                span.SetTag(Datadog.Trace.Tags.AzureAppServicesResourceId, AzureAppServices.Metadata.ResourceId);
-                span.SetTag(Datadog.Trace.Tags.AzureAppServicesInstanceId, AzureAppServices.Metadata.InstanceId);
-                span.SetTag(Datadog.Trace.Tags.AzureAppServicesInstanceName, AzureAppServices.Metadata.InstanceName);
-                span.SetTag(Datadog.Trace.Tags.AzureAppServicesOperatingSystem, AzureAppServices.Metadata.OperatingSystem);
-                span.SetTag(Datadog.Trace.Tags.AzureAppServicesRuntime, AzureAppServices.Metadata.Runtime);
-                span.SetTag(Datadog.Trace.Tags.AzureAppServicesExtensionVersion, AzureAppServices.Metadata.SiteExtensionVersion);
+                span.Tags.SetTag(Datadog.Trace.Tags.AzureAppServicesSiteName, AzureAppServices.Metadata.SiteName);
```

Comment:
Just for my understanding, what is the rationale behind not being able to set tags on a `IsFinished` span through `Span.SetTag`, but being able to actually set those tags through the `ITags.SetTag` call?

Is that something that is just a side-effect of the APIs or is there an actual reason for that?

Reply:
I don't know for sure to be honest. I assumed it was done to prevent users to set tags too late in the process and the `Tags` property isn't accessible publicly.

Reply:
Okay makes sense to just restrict that from being publicly accessible, but still allow for tags to be modified internally.